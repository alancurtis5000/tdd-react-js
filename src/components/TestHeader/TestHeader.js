import React from 'react';

export default function TestHeader({ text }) {
  return (
    <div>
      <h1 data-testid="header-text" className="specific-class">
        {text}
      </h1>
    </div>
  );
}
