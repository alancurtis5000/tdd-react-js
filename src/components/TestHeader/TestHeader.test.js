import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestHeader from './TestHeader';

describe('TestHeader', () => {
  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<TestHeader text="hello" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('inserts text in h1', () => {
    const { getByTestId, getByText } = render(<TestHeader text="hello" />);
    expect(getByTestId('header-text')).toHaveTextContent('hello');
    expect(getByText('hello')).toHaveClass('specific-class');
  });
});
