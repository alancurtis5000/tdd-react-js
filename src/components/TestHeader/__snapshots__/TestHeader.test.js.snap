// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestHeader renders 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="specific-class"
      data-testid="header-text"
    >
      hello
    </h1>
  </div>
</DocumentFragment>
`;
